Source/x11/XGServerWindow.m

```diff
@@ -2815,6 +2815,24 @@ - (void) orderwindow: (int)op : (int)otherWin : (int)winNum
        */
       if ((window->win_attrs.window_style & NSIconWindowMask) != 0)
 	{
+          NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+          if (op != NSWindowOut && window->map_state == IsUnmapped &&
+              [[defaults objectForKey: @"autolaunch"] isEqualToString:@"YES"])
+ 	    {
+              XEvent ev;
+
+              // Inform WM to ignore focus events
+              ev.xclient.type = ClientMessage;
+              ev.xclient.message_type = XInternAtom(dpy,"WM_IGNORE_FOCUS_EVENTS", False);
```

Comment:
In GNUstep back I did remove all the local calls to XInternAtom and get all the required atoms once at the start of the application, see _setupRootWindow. These atoms are defined in the file XGGeneric.h.
The only place my Google search found this specific atom "WM_IGNORE_FOCUS_EVENTS" was in WindowMaker source code. Is it only supported for this window manager? I also didn't find it in my X Windows reference book. Where does this come from?

Reply:
Yes it's WindowMaker only atom (defined here: https://repo.or.cz/wmaker-crm.git/blob/HEAD:/src/startup.c#l399).
Anyway this `if (generic.flags.useWindowMakerIcons == 1)` block (starts at line 2808) will be invoked only for WindowMaker. I suppose `useWindowMakerIcons` flag make sense only for WindowMaker.